Source/WebCore/Modules/fetch/FetchHeaders.cpp

```diff
@@ -57,9 +57,25 @@ static ExceptionOr<bool> canWriteHeader(const String& name, const String& value,
     return true;
 }
 
-static ExceptionOr<void> appendToHeaderMap(const String& name, const String& value, HTTPHeaderMap& headers, FetchHeaders::Guard guard)
+static ExceptionOr<void> appendSetCookie(const String& value, Vector<String>& setCookieValues, FetchHeaders::Guard guard)
+{
+    if (!isValidHTTPHeaderValue((value)))
+        return Exception { TypeError, makeString("Header 'Set-Cookie' has invalid value: '", value, "'") };
+    if (guard == FetchHeaders::Guard::Immutable)
+        return Exception { TypeError, "Headers object's guard is 'immutable'"_s };
+
+    if (guard == FetchHeaders::Guard::None)
+        setCookieValues.append(value);
+
+    return { };
+}
+
+static ExceptionOr<void> appendToHeaderMap(const String& name, const String& value, HTTPHeaderMap& headers, Vector<String>& setCookieValues, FetchHeaders::Guard guard)
 {
     String normalizedValue = stripLeadingAndTrailingHTTPSpaces(value);
+    if (equalIgnoringASCIICase(name, "Set-Cookie"_str))
```

Comment:
Why camel-cased `"Set-Cookie"_str`? Wouldn't it be better to use lower cased ASCIILiteral `"set-cookie"_s`?